src/sbvr-api/sbvr-utils.coffee

```diff
@@ -768,18 +785,42 @@ runChangeSet = (req, res, tx) ->
 # This function compiles the sql query of a request which has been deferred
 updateBinds = (env, request) ->
 	if request._defer
+		modifiedFields = AbstractSQLCompiler.getModifiedFields(request.abstractSqlQuery)
 		request.odataBinds = _.map request.odataBinds, ([tag, id]) ->
 			if tag == 'ContentReference'
-				ref = env.get(id)
-				if _.isUndefined(ref?.body?.id)
+				result = env.results.get(id)
+				validation = env.validation.get(id)
+				if _.isUndefined(result?.body?.id) or _.isUndefined(validation?.modifiedFields)
 					throw uriParser.BadRequestError('Reference to a non existing resource in Changeset')
 				else
-					uriParser.parseId(ref.body.id)
+					# We also have to invalidate the affected fields on the referenced query
+					validation.modifiedFields = invalidate(validation.modifiedFields, modifiedFields)
+					env.validation.set(id, validation)
+					uriParser.parseId(result.body.id)
 			else
 				[tag, id]
 		request.sqlQuery = memoizedCompileRule(request.abstractSqlQuery)
+		request.modifiedFields = modifiedFields
 	return request
 
+
+# Removes from oldFields every field appearinf in newFields
+invalidate = (oldFields, newFields) ->
+	if _.isArray(oldFields)
+		return _.map(oldFields, _.partial(invalidate, _, newFields))
+	else if _.isArray(newFields)
+		return _.reduce(newFields, invalidate, oldFields)
+	else if _.isUndefined(newFields)
```

Comment:
I guess that this could be the first thing to check.
Also I guess that all these could be `return ... if ...` statements, which would remove all the `else`s, but I'm not sure which we prefer.